rest/data/project.go

```diff
@@ -96,6 +96,16 @@ func (sc *DBProjectConnector) VerifyUniqueProject(name string) error {
 	return nil
 }
 
+// GroupProjectsByRepo takes in an array of projects and groups them in a map based on the repo they are part of
+func (sc *DBProjectConnector) GroupProjectsByRepo(projects []model.ProjectRef) map[string][]model.ProjectRef {
+	groupedProject := make(map[string][]model.ProjectRef)
+	for _, project := range projects {
+		repoProjects := groupedProject[project.Repo]
```

Comment:
Theoretically there can be multiple repos with the same name; it's the owner/repo pair that's guaranteed to be unique, or the RepoRefId (This may be better because we don't want to group projects that aren't defaulting to repo; it implies that they have a connection to each other where they do not).